python/src/tests/tasks/test_list_genes.py

```diff
@@ -8,88 +8,126 @@
-        res = ListGenes(self.correct_request_skeleton, self._adata).compute()
-        res = res[0].result
-        res = json.loads(res)
-
-        assert isinstance(res["total"], int)
+    def test_descending(self):
+        res = ListGenes(self.correct_desc).compute()
+        res = json.loads(res[0].result)
+        assert res == self.correct_response["desc_20"]
 
-    def test_result_object_has_total_results_which_is_list(self):
-        res = ListGenes(self.correct_request_skeleton, self._adata).compute()
-        res = res[0].result
-        res = json.loads(res)
+    def test_ascending(self):
+        res = ListGenes(self.correct_asc).compute()
+        res = json.loads(res[0].result)
+        assert res == self.correct_response["asc_20"]
 
-        assert isinstance(res["rows"], list)
+    def test_by_names(self):
+        res = ListGenes(self.correct_names).compute()
+        res = json.loads(res[0].result)
+        assert res == self.correct_response["names_desc"]
 
     def test_list_gene_selected_fields_appear_in_all_results(self):
-        res = ListGenes(self.correct_request_skeleton, self._adata).compute()
+        res = ListGenes(self.correct_desc).compute()
         res = res[0].result
         res = json.loads(res)
 
         for data in res["rows"]:
             for field in data.keys():
-                assert field in self.correct_request_skeleton["body"]["selectFields"]
+                assert field in self.correct_desc["body"]["selectFields"]
 
     def test_list_gene_has_appropriate_number_of_results(self):
-        res = ListGenes(self.correct_request_skeleton, self._adata).compute()
+        res = ListGenes(self.correct_filter).compute()
         res = res[0].result
         res = json.loads(res)
         res = res["rows"]
 
-        assert len(res) <= self.correct_request_skeleton["body"]["limit"]
+        assert len(res) <= self.correct_names["body"]["limit"]
 
     def test_filter_contains_pattern_gets_applied_to_results(self):
-        task_def = self.correct_request_skeleton["body"]
-        task_def["geneNamesFilter"] = "%LIN%"
-        res = ListGenes(self.correct_request_skeleton, self._adata).compute()
+        res = ListGenes(self.correct_filter).compute()
         res = json.loads(res[0].result)
-
         for row in res["rows"]:
-            assert "LIN" in row["gene_names"]
+            assert "lin".casefold() in row["gene_names"].casefold()
 
     def test_filter_starts_with_pattern_gets_applied_to_results(self):
-        task_def = self.correct_request_skeleton["body"]
-        task_def["geneNamesFilter"] = "LIN%"
-        res = ListGenes(self.correct_request_skeleton, self._adata).compute()
+        res = ListGenes(self.correct_startswith).compute()
         res = json.loads(res[0].result)
 
         for row in res["rows"]:
-            assert row["gene_names"].startswith("LIN")
-"""
\ No newline at end of file
+            assert row["gene_names"].casefold().startswith("LIN".casefold())
```

Comment:
```suggestion
            assert row["gene_names"].lower().startswith("lin")
```
Use hardcoded `"lin"` or `"LIN".lower()` for both this and previous test for consistency